bank/src/main/java/org/example/Bank/Bank.java

```diff
@@ -0,0 +1,192 @@
+
+
+//util pack import
+
+import org.example.Account.Account;
+import org.example.Account.savingAccount;
+import org.example.Exception.UnknownException;
+import org.example.Exception.WithdrawException;
+
+import java.math.BigDecimal;
+import java.text.DecimalFormat;
+import java.util.HashMap;
+import java.util.InputMismatchException;
+import java.util.Scanner;
+
+
+//exception part
+
+/*
+ * 			계좌 클래스에서 구현한 기본 클래스를 이용하여 계좌생성, 출금, 입금, 송금 메서드를 구현합니다. 메서드 내부적으로 입력값을 받는 액션이 있습니다.
+ *
+			적금 계좌는 적금 계좌는 잔액이 목표 금액(%s원) 이상이어야 출금 가능하도록 상속받은 출금 메서드를 조금 다르게 구현해줍니다.
+
+ 모든 계좌를 관리하기 위한 중앙은행 클래스를 만듭니다. 계좌를 배열에 관리하기 위해 ArrayList를 활용합니다.
+계좌별 이자 정책을 관리하기 위해 HashMap을 활용하고, key에 category, value에 각 category의 InterestCalculator 인스턴스 를 담습니다.
+ */
+
+public class Bank {
+    protected static Scanner scanner = new Scanner(System.in);
+    protected static int seq = 0;
+    public static DecimalFormat df = new DecimalFormat("#,###");
+
+    ///이자계산을 위한 해시맵
+
+
+    private HashMap<String, InterestCalculator> interestCalculators = new HashMap<>();
+
+    public Bank(){
+        //이자계산기 11
+        interestCalculators.put("N", (InterestCalculator) new BasicInterestCalculator());
+        interestCalculators.put("S", new SavingInterestCalculator());
+    }
+
+
+    //출금 메서드
+    public void withdraw() throws WithdrawException, UnknownException {
+
+        //1. 계좌번호를 입력하고,맞는지 확인한다.
+
+        Account account = null;
+        while(true) {
+            String accountNum = askInput("\n 출금하려는 계좌의 번호를 입력하세요 : ", "");
+
+            try {
+                account = findAccount(accountNum);
+
+
+                if(account.getCategory().equals("S")) {
+                    //savingAccont로 type casting 해줘야하는 이유???
+                    new SavingBank().withdraw((savingAccount) account);
+                }
+                //목적 달성하면 break.
+                break;
+
+
+            } catch(WithdrawException err) {
+                System.out.println(err.getMessage());
+                return;
+            }
+            catch(Exception e) {
+                System.out.println(e.getMessage());
+                break;
+            }
+
+        }
+
+        // 출금처리
+        while(true){
+            try{
+                //askinput 함수에 명시적으로 숫자타입이라고 알려주는거임
+                BigDecimal amount = askInput("\n출금할 금액을 입력하세요.", BigDecimal.ZERO);
+                BigDecimal result;
+
+                //이자 계산
+                BigDecimal interest = interestCalculators.get(account.getCategory()).getInterest(account.getBalance());
+                result = account.withdraw(amount);
+                System.out.printf("\n출금액 %s원과 이자 %s원이 정상적으로 출금되었습니다.\n", df.format(result), df.format(interest));
+                break;
+            }catch(WithdrawException we) {
+                throw we;
+            }catch (Exception e){
+                throw new WithdrawException(e.getMessage());
+            }
+        }
+    }
+
+
+
+    //계좌 finding
+    public Account findAccount(String accNo)  {
```

Comment:
변수명을 줄임말로 사용하는 것은 좋지  않습니다.
```suggestion
    public Account findAccount(String accountNumber)  {
```